defaults/security-ents.go

```diff
@@ -0,0 +1,162 @@
+package defaults
+
+import (
+	"fmt"
+
+	"github.com/docker/libentitlement/defaults/osdefs"
+	"github.com/docker/libentitlement/entitlement"
+	"github.com/docker/libentitlement/secprofile"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+const (
+	securityDomain = "security"
+)
+
+const (
+	// SecurityConfinedEntFullID is the ID for the security.confined entitlement
+	SecurityConfinedEntFullID = securityDomain + ".confined"
+	// SecurityViewEntFullID is the ID for the security.view entitlement
+	SecurityViewEntFullID = securityDomain + ".view"
+	// SecurityAdminEntFullID is the ID for the security.admin entitlement
+	SecurityAdminEntFullID = securityDomain + ".admin"
+	// SecurityMemoryLockFullID is the ID for the security..memory-lock entitlement
+	SecurityMemoryLockFullID = securityDomain + ".memory-lock"
+)
+
+var (
+	securityConfinedEntitlement   = entitlement.NewVoidEntitlement(SecurityConfinedEntFullID, securityConfinedEntitlementEnforce)
+	securityViewEntitlement       = entitlement.NewVoidEntitlement(SecurityViewEntFullID, securityViewEntitlementEnforce)
+	securityAdminEntitlement      = entitlement.NewVoidEntitlement(SecurityAdminEntFullID, securityAdminEntitlementEnforce)
+	securityMemoryLockEntitlement = entitlement.NewVoidEntitlement(SecurityMemoryLockFullID, securityMemoryLockEnforce)
+)
+
+func securityConfinedEntitlementEnforce(profile secprofile.Profile) (secprofile.Profile, error) {
+	if profile.GetType() != secprofile.OCIProfileType {
+		return nil, fmt.Errorf("%s not implemented for non-OCI profiles", SecurityConfinedEntFullID)
+	}
+
+	ociProfile, ok := profile.(*secprofile.OCIProfile)
+	if !ok {
+		return nil, fmt.Errorf("%s: error converting to OCI profile", SecurityConfinedEntFullID)
+	}
+
+	capsToRemove := []string{"CAP_MAC_ADMIN", "CAP_MAC_OVERRIDE", "CAP_DAC_OVERRIDE", "CAP_DAC_READ_SEARCH",
+		"CAP_SETPCAP", "CAP_SETFCAP", "CAP_SETUID", "CAP_SETGID", "CAP_SYS_PTRACE", "CAP_FSETID", "CAP_SYS_MODULE",
+		"CAP_SYSLOG", "CAP_SYS_RAWIO", "CAP_SYS_ADMIN", "CAP_LINUX_IMMUTABLE",
```

Comment:
followup PR: we should probably validate all capability and syscall strings. I couldn't find any good golang package primitives so we may have to follow something like this: https://github.com/moby/tool/pull/93/files